src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -44,12 +63,20 @@ public Drivetrain() {
   @Override
   public void periodic(){
     // TODO 2.2.5: Update odometry
-
-    // TODO 1.2.2: Call tankDrive()
+    tankDrive(Robot.driver.getRawLeftTranslation(), Robot.driver.getRawRightTranslation());
+    
+    drivePose.update(getGyroAngle(), getLeftPosition(), getRightPosition());
 
     // TODO 3.1.1: Remove all of the tank drive code in this method
 
+
     // TODO 2.1.3: Update sim if in simulation
+
+    if (RobotBase.isReal()) {
+
+    }else{
+      driveSim.update(Constants.LOOP_TIME);
```

Comment:
You do not need an empty if statement. You can either use ! or RobotBase.isSimulation().